src/simulator/views/map/display/solid_colour_map_display.py

```diff
@@ -65,7 +70,8 @@ def get_tracked_data(self) -> List[Tracked]:
         return [self.pts] if isinstance(self.pts, Tracked) else []
 
     def update_cube(self, p: Point) -> None:
-        if self._map.size.n_dim == 2:
+        if self.__point_dim != 3:
```

Comment:
this may cause bugs with nd later

Reply:
No way around it, we have to trust that the user either only deals with 2D or only with 3D points.

Reply:
The old fix was actually causing bugs, this new one doesn't.

Reply:
really wtf that is definitely sus behaviour then

Reply:
?

Reply:
r u saying that n_dim == 2 doesnt work but __point_dim != 3 works for some reason? thats odd

Reply:
Yes because __point_dim only relies on the type of points the user uses, so much less surprises. This guarantees correct point checking. Users may start using 3D points (such as myself). I personally find this default much more intuitive and robust.